ui/ui/predict_ui.py

```diff
@@ -0,0 +1,126 @@
+                    
+                    1. Select the input values for the predictions 
+                    2. Select the output values to be shown in the results
+                    3. Click on the `Predict` button
+                    
+                    4. The results will be shown in the main page
+
+                    > Note: Make sure to train the models before performing prediction, more info can be found on the   
+                    on the training page.  
+                    > Note: Be patient. The operations might take some time.    
+                    """)
+
+        if 'prediction_csv' not in st.session_state:
+            st.session_state['prediction_csv'] = None
+        st.write(st.session_state.prediction_csv)
+
+        if 'prediction_results' not in st.session_state:
+            st.session_state['prediction_results'] = ""
+        st.markdown(st.session_state.prediction_results)
+
+
+    def add_form_element(self, element):
+        if element["type"] == "text_input":
+            st.text_input(element["name"], key=element["name"], placeholder=element["placeholder"])
+        elif element["type"] == "toggle":
+            st.toggle(element["name"], key=element["name"], value=element["value"])
+        else:
+            st.text(f"Unknown element type {element}")
+
+    def show_sidebar(self):
+        with (st.sidebar):
+            st.header("Prediction configuration:")
+            with st.form("Configuration"):
+                st.subheader("Select input configuration:",
+                             help="For each of the input features, select a specific value or"
+                                  "use-multi selection to predict for all the values")
+                for input_field_value in self.input_fields.values():
+                    self.add_form_element(input_field_value)
+
+                st.divider()
+                st.subheader("Select output configuration:",
+                             help="For each of the output features, select which features to present")
+
+                for output_field_value in self.output_fields.values():
+                    self.add_form_element(output_field_value)
+
+                st.form_submit_button("Predict", on_click=self.on_predict)
+
+    def on_predict(self):
+        from config import config
+
+        logger.debug(f"on_predict")
+        command_template = Template(config["actuation_templates"]["predict"])
+        logger.debug(f"command_template: {command_template}")
+
+        # Substitute values
+        command = command_template.substitute(title=self.title)
+
+        result = execute_command(command)
+        logger.debug(f"on_predict result: {result}")
+
+
+        st.session_state['prediction_results'] = f"""
+            ## Execution of `prediction` is completed!  
+
+            The results are:
+            
+            {result}
+            """
+
+        csv_result = pd.read_csv("../examples/MLCommons/ARISE-predictions/all-predictions.csv")
```

Comment:
result should be read from the user's data folder, which is not necessarily MLCommons

Reply:
Everything is read now in the configuration file ("config.yaml")